spec/models/population_spec.rb

```diff
@@ -151,6 +151,51 @@
     end
   end
 
+  describe '.new_with_defaults' do
+    subject(:new_population) { described_class.new_with_defaults(location: location, date: date) }
+
+    let(:location) { create(:location, :prison) }
+    let(:date) { Date.today }
+
+    context 'with a previous population record for same location' do
+      let!(:previous_population) { create(:population, location: location, date: date - 1.day) }
+
+      it 'populates details from previous record' do
+        expect(new_population).to have_attributes({
+          id: nil,
+          location_id: location.id,
+          date: date,
+          operational_capacity: previous_population.operational_capacity,
+          usable_capacity: previous_population.usable_capacity,
+          unlock: previous_population.unlock,
+          bedwatch: previous_population.bedwatch,
+          overnights_in: previous_population.overnights_in,
+          overnights_out: previous_population.overnights_out,
+          out_of_area_courts: previous_population.out_of_area_courts,
+          discharges: previous_population.discharges,
+        })
+      end
+    end
+
+    context 'without a previous population record' do
```

Comment:
can you also add a test for checking that it selects the latest previous population?